src/optimizer/plugins/RemoveGeneratedTextRulesOptimizer.ts

```diff
@@ -32,5 +33,45 @@ export default {
       }
       pullAt(resource.rules as ExportableRule[], rulesToRemove);
     });
+
+    [...pkg.instances].forEach(instance => {
+      const rulesToRemove: number[] = [];
+      if (hasGeneratedText(instance)) {
+        instance.rules.forEach((rule: ExportableRule, i: number) => {
+          if (rule instanceof ExportableAssignmentRule && rule.path.match(/^text\./)) {
+            rulesToRemove.push(i);
+          }
+        });
+      }
+      pullAt(instance.rules as ExportableRule[], rulesToRemove);
```

Comment:
I think maybe we're using `pullAt` more than necessary in our optimizers now.  I think it makes sense to use when we might have multiple loops or other reasons for not messing w/ indexes while we process rules, but in cases like above, can't we just get away with:
```ts
if (hasGeneratedText(instance)) {
  instance.rules = instance.rules.filter(rule => {
    return !(rule instanceof ExportableAssignmentRule && rule.path.match(/^text\./));
  });
}
```
or using lodash [remove](https://lodash.com/docs/4.17.15#remove):
```ts
if (hasGeneratedText(instance)) {
  remove(instance.rules, rule => rule instanceof ExportableAssignmentRule && rule.path.match(/^text\./));
}
```

Reply:
Also, I think the same would go for lines 22 - 34 above.

Reply:
Cleaned up in https://github.com/FHIR/GoFSH/pull/45/commits/de97ccec2574c3fdcd07d9478992e1fecb91ff11